packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.test.jvm
+
+import io.realm.kotlin.Realm
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.entities.link.Child
+import io.realm.kotlin.entities.link.Parent
+import io.realm.kotlin.internal.platform.singleThreadDispatcher
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.test.util.use
+import kotlinx.coroutines.CloseableCoroutineDispatcher
+import kotlinx.coroutines.async
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Realm tests that are specific to the JVM platform (both Desktop and Android).
+ */
+class RealmTests {
+
+    @Test
+    fun cleanupDispatcherThreadsOnClose() = runBlocking {
+        val tmpDir = PlatformUtils.createTempDir()
+        val startingThreadCount = Thread.activeCount()
```

Comment:
Be careful this only returns the number of threads within the current `ThreadGroup`